### ISSUE TITLE
Logback to log4j2 support

### ISSUE THREAD

**Issue by Misha123456** [openrewrite/rewrite-logging-frameworks#97]
I was reading the documentation [here](https://docs.openrewrite.org/reference/recipes/java/logging/logback) and can see log4j2 to logback conversion is supported. 
But there is no way to convert vice versa (from logback to log4j2).

Are there plans to implement this in the future?

**Comment by ppkarwasz** [comment]
@timtebeek,

Could you assign this issue to me?

I would split it into two independent recipes:

- "SLF4J to Log4j API", which mostly what library developers are interested in,
- "Logback to Log4j Core", which would be targeted to application developers and just switches dependencies and configuration.

**Comment by timtebeek** [comment]
Sure sounds good; Thanks for taking this on!

**Comment by timtebeek** [comment]
We recently completed quite a few recipes to move from JUL to SLF4J; those are now under:
https://github.com/openrewrite/rewrite-logging-frameworks/tree/main/src/main/java/org/openrewrite/java/logging/slf4j

I suspect that the approach taken there could be largely followed here: Mostly use declarative or Refaster recipes, and complement those with explicit recipes as needed, which wouldn't need much.

**Comment by ppkarwasz** [comment]
We have recently revamped our documentation with:

- A [Migrating from SLF4J](https://logging.staged.apache.org/log4j/2.x/migrate-from-slf4j.html) guide, which points to the existing OpenRewrite recipe.
- A [Migrating from Logback](https://logging.staged.apache.org/log4j/2.x/migrate-from-logback.html) guide, which will point to the Logback to Log4j Core recipe once it is ready.

Unlike the migration from JUL to SLF4J, which requires mostly code changes, migrating from Logback to Log4j Core requires:

1. A migration of runtime dependencies, which can be easily done with the current declarative Maven/Gradle recipes.
2. A migration of the configuration file.

The latter requires most of the work, since the configuration options of every standard Logback plugin must be matched to the corresponding options of a Log4j Core plugin. Starting October 2024, I plan to add support for a Logback-to-Log4j Core configuration conversion to the `log4j-config-converter` artifact introduced in https://github.com/apache/logging-log4j-transform/pull/93. This artifact can then be added as a dependency of the OpenRewrite recipe.

**Comment by timtebeek** [comment]
Nice to see, thanks! Spotted this note in the docs that ought to be an easy recipe for your users too.
> Note that `LogManager.getLogger(Foo.class)` can be simplified as `LogManager.getLogger()`, if Foo is the enclosing class of the field.
